wallet/src/account.rs

```diff
@@ -134,18 +128,15 @@ impl Wallet {
     /// The returned value is the value we expect to see at some point on
     /// chain once all transactions are on chain confirmed.
     pub fn unconfirmed_value(&self) -> Option<Value> {
-        let (k, s, _) = self.state.last_state();
-        let (kk, _) = self.state.confirmed_state();
+        let s = self.state.last_state();
 
-        if k == kk {
-            None
-        } else {
-            Some(s.value)
-        }
+        Some(s)
+            .filter(|s| !s.is_confirmed())
+            .map(|s| s.state().value)
```

Comment:
A bit less straightforward than the plain `if s.is_confirmed() { ... } else { ... }`, but I suppose it works just as well.